casr/src/bin/casr-java.rs

```diff
@@ -140,7 +141,30 @@ fn main() -> Result<()> {
 
     if let Ok(crash_line) = JavaStacktrace::parse_stacktrace(&report.stacktrace)?.crash_line() {
         report.crashline = crash_line.to_string();
-        if let CrashLine::Source(debug) = crash_line {
+        if let CrashLine::Source(mut debug) = crash_line {
+            // Modify DebugInfo to find sources
+            let source_dirs: Vec<PathBuf> = if let Ok(sources) = std::env::var("CASR_SOURCE_DIRS") {
+                sources
+                    .split(':')
+                    .map(PathBuf::from)
+                    .filter(|x| x.is_dir())
+                    .collect()
+            } else {
+                Vec::new()
+            };
+
+            if let Some(file) = source_dirs.iter().find_map(|dir| {
+                WalkDir::new(dir)
+                    .into_iter()
+                    .flatten()
+                    .map(|e| e.into_path())
+                    .filter(|e| e.is_file())
+                    .filter(|e| e.extension().is_some() && e.extension().unwrap() == "java")
+                    .find(|x| x.file_name().unwrap().to_str().unwrap().eq(&debug.file))
```

Comment:
shouldn't we also get debug.file.file_name() to enable this search for arbitrary language (not only Java)?

Reply:
We use this search only for java. In java `debug.file` contains only filename.

Reply:
debug.file is actually a full path to that file

Reply:
It contains just a file name only in Java. However, we may write a more general case code.

Reply:
debug.file -> debug.file.file_name()

Reply:
done